featurebyte/service/offline_store_feature_table_manager.py

```diff
@@ -53,54 +156,48 @@ def __init__(
         self.feast_registry_service = feast_registry_service
         self.feature_list_service = feature_list_service
 
-    async def handle_online_enabled_feature(self, feature: FeatureModel) -> None:
+    async def handle_online_enabled_features(self, features: List[FeatureModel]) -> None:
         """
-        Handles the case where a feature is enabled for online serving by updating all affected
+        Handles the case where features are enabled for online serving by updating all affected
         feature tables.
 
         Parameters
         ----------
-        feature: FeatureModel
-            Model of the feature to be enabled for online serving
+        features: List[FeatureModel]
+            Features to be enabled for online serving
         """
-        feature_entities = []
-        async for entity_model in self.entity_service.list_documents_iterator(
-            query_filter={"_id": {"$in": feature.entity_ids}}
-        ):
-            feature_entities.append(entity_model)
-
-        # FIXME: cleanup this logic once integrate this with feast feature store
-        if feature.offline_store_info is None:
-            feature.initialize_offline_store_info(
-                entity_id_to_serving_name={
-                    entity.id: entity.serving_names[0] for entity in feature_entities
-                }
-            )
-
-        offline_store_info = feature.offline_store_info
-        assert offline_store_info is not None, "Offline store info should not be None"
-        offline_ingest_graphs = offline_store_info.extract_offline_store_ingest_query_graphs()
+        ingest_graph_container = await OfflineIngestGraphContainer.build(
+            self.entity_service, features
+        )
 
-        for offline_ingest_graph in offline_ingest_graphs:
+        for (
+            offline_store_table_name,
+            offline_store_table_features,
+        ) in ingest_graph_container.iterate_features_by_table_name():
             feature_table_dict = await self._get_compatible_existing_feature_table(
-                offline_ingest_graph=offline_ingest_graph
+                table_name=offline_store_table_name,
             )
 
             if feature_table_dict is not None:
                 # update existing table
                 feature_ids = feature_table_dict["feature_ids"][:]
```

Comment:
Could we use set here?

Reply:
Sure, will update to use set in a follow up.